cmds/core/find/find.go

```diff
@@ -44,58 +69,67 @@ func init() {
 	}
 }
 
-func main() {
+func (c *command) run() error {
 	fileTypes := map[string]os.FileMode{
 		"f":         0,
 		"file":      0,
 		"d":         os.ModeDir,
 		"directory": os.ModeDir,
 	}
 
-	flag.Parse()
-	a := flag.Args()
-	if len(a) != 1 {
+	if len(c.args) != 1 {
 		flag.Usage()
 	}
-	root := a[0]
+	root := c.args[0]
 
 	var mask, mode os.FileMode
-	if *perm != -1 {
+	if c.params.perm != -1 {
 		mask = os.ModePerm
-		mode = os.FileMode(*perm)
+		mode = os.FileMode(c.params.perm)
 	}
-	if *fileType != "" {
-		intType, ok := fileTypes[*fileType]
+	if c.params.fileType != "" {
+		intType, ok := fileTypes[c.params.fileType]
 		if !ok {
 			var keys []string
 			for key := range fileTypes {
 				keys = append(keys, key)
 			}
-			log.Fatalf("%v is not a valid file type\n valid types are %v", *fileType, strings.Join(keys, ","))
+			return fmt.Errorf("%v is not a valid file type\n valid types are %v", c.params.fileType, strings.Join(keys, ","))
 		}
 		mode |= intType
 		mask |= os.ModeType
 	}
 
 	debugLog := func(string, ...interface{}) {}
-	if *debug {
+	if c.params.debug {
 		debugLog = log.Printf
 	}
 	names := find.Find(context.Background(),
 		find.WithRoot(root),
 		find.WithModeMatch(mode, mask),
-		find.WithFilenameMatch(*name),
+		find.WithFilenameMatch(c.params.name),
 		find.WithDebugLog(debugLog),
 	)
+
 	for l := range names {
 		if l.Err != nil {
-			fmt.Fprintf(os.Stderr, "%v: %v\n", l.Name, l.Err)
+			fmt.Fprintf(c.strerr, "%s: %v\n", l.Name, l.Err)
 			continue
 		}
-		if *long {
-			fmt.Printf("%s\n", l)
+		if c.params.long {
+			fmt.Fprintf(c.stdout, "%s\n", l)
 			continue
 		}
-		fmt.Printf("%s\n", l.Name)
+		fmt.Fprintf(c.stdout, "%s\n", l.Name)
+	}
+
+	return nil
+}
+
+func main() {
```

Comment:
This is nice structure, as it will allow parallel testing. One further question: can we remove the global flags to main()? golangci-lint is now generating warnings for all global variables, including flags, and looking to the future it might be best to remove them where we can. Thanks.

Reply:
Sure we can, even we will take a coverage hit :)

Reply:
one way NOT to take a coverage hit: have a function which runs it and accepts args as ...string. 
Then you can do this
err := flag.NewFlagSet("find", whatever).Parse(args...)
so no use of globals at all, and main becomes this
func main() {
if err := run(os.Args...); err != nil {
log.Fatal(err)
}
}

and we can do even more testing.
wdyt?

Reply:
I'm a bit skeptical by introducing NewFlagSet for 5 lines of code coverage, and partially because I don't get how it will help. It's not clear for me where these lines will go:
```go
debug    = flag.Bool("d", false, "Enable debugging in the find package")
...
```
When they are global the coverage is good, but when they are in main it's not the case. For simple flags parsing i would go with just main and don't bother with coverage.

As alternative some time ago i used function like this, if some flag is mandatory we can extend it will error and panic right away.
```go
type flags struct {
	name string
	age  int
}

func parseFlags() flags {
	f := flags{}
	flag.StringVar(&res.name, "name", "default", "flag name")
	flag.IntVar(&res.age, "age", 42, "flag age")
	flag.Parse()
	return f
}
```